ludwig/hyperopt.py

```diff
@@ -0,0 +1,512 @@
+#! /usr/bin/env python
+# coding=utf-8
+# Copyright (c) 2019 Uber Technologies, Inc.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+# ==============================================================================
+from __future__ import absolute_import
+from __future__ import division
+from __future__ import print_function
+
+import argparse
+import logging
+from typing import Any, Dict, List
+import yaml
+
+from ludwig.constants import EXECUTOR, HYPEROPT, STRATEGY
+from ludwig.contrib import contrib_command
+from ludwig.models.hyperopt import get_build_hyperopt_strategy, get_build_hyperopt_executor
+from ludwig.globals import LUDWIG_VERSION, is_on_master, set_on_master
+from ludwig.predict import (predict, print_test_results,
+                            save_prediction_outputs, save_test_statistics)
+from ludwig.utils.defaults import default_random_seed, merge_with_defaults
+from ludwig.utils.misc import set_default_value, set_default_values
+from ludwig.utils.print_utils import logging_level_registry, print_ludwig
+
+
+logger = logging.getLogger(__name__)
+
+
+class HyperoptStrategy:
+    def __init__(self, strategy: Dict[str, Any], parameters: Dict[str, Any]) -> None:
```

Comment:
the `strategy: Dict[str, Any]` is not needed. Different objects implementing this abstract class will have different strategy parameters.